sql/core/src/test/resources/sql-tests/inputs/operators.sql

```diff
@@ -95,3 +95,9 @@ select width_bucket(5.35, 0.024, 10.06, null);
 select width_bucket(5.35, 0.024, 10.06, -5);
 select width_bucket(5.35, 0.024, 10.06, 9223372036854775807L); -- long max value
 select width_bucket(5.35, 0.024, 10.06, 9223372036854775807L - 1);
+
+-- special floating values
+select double('NaN') = double('NaN'), double('NaN') >= double('NaN'), double('NaN') <= double('NaN');
+select double('NaN') > double("+Infinity"), double("+Infinity") < double('NaN'), double('NaN') > double('NaN');
+select 0.0 = -0.0, 0.0 >= -0.0, 0.0 <= -0.0;
+select 0.0 > -0.0, 0.0 < -0.0;
```

Comment:
This doesn't add a new test coverage because this has the same result with Spark 3.0.0.
```
spark-sql> select version();
3.0.0 3fdfce3120f307147244e5eaf46d61419a723d50

spark-sql> select double('NaN') = double('NaN'), double('NaN') >= double('NaN'), double('NaN') <= double('NaN');
true	true	true

spark-sql> select double('NaN') > double("+Infinity"), double("+Infinity") < double('NaN'), double('NaN') > double('NaN');
true	true	false

spark-sql> select 0.0 = -0.0, 0.0 >= -0.0, 0.0 <= -0.0;
true	true	true

spark-sql> select 0.0 > -0.0, 0.0 < -0.0;
false	false
```

Reply:
We can remove these if we cannot reproduce the regression in SQL layer. Otherwise, this may be misleading.

Reply:
literals are OK. I'll update the tests to use columns.

Reply:
I'm removing them. I can't reproduce the bug with temp view as there are optimizations around literals. I'll add an end-to-end test in `DataFrameSuite`.

Reply:
Thanks!